src/sentry/utils/sdk_crashes/event_stripper.py

```diff
@@ -37,11 +75,37 @@ def strip_event_data(event: Event, sdk_crash_detector: SDKCrashDetector) -> Even
     return new_event_data
 
 
-def _filter_contexts(pair: Tuple[str, Any]) -> bool:
-    key, _ = pair
-    if key in {"os", "device"}:
-        return True
-    return False
+def _strip_event_data_with_allowlist(
+    data: Mapping[str, Any], allowlist: Optional[Mapping[str, Any]]
+) -> Optional[Mapping[str, Any]]:
+    """
+    Recursively traverses the data and only keeps values based on the allowlist.
+    """
+    if allowlist is None:
+        return None
+
+    stripped_data: Dict[str, Any] = {}
+    for data_key, data_value in data.items():
+        allowlist_for_data = allowlist.get(data_key)
+        if allowlist_for_data is None:
+            continue
+
+        if isinstance(allowlist_for_data, Allow):
+            allowed = allowlist_for_data
+
+            if allowed is Allow.SIMPLE_TYPE and isinstance(data_value, (str, int, float, bool)):
+                stripped_data[data_key] = data_value
+            elif allowed is Allow.ALL:
+                stripped_data[data_key] = data_value
+            else:
+                continue
+
+        elif isinstance(data_value, Mapping):
+            stripped_data[data_key] = _strip_event_data_with_allowlist(
+                data_value, allowlist_for_data
+            )
```

Comment:
Can this function also descend into lists? I imagine that would be necessary.

Reply:
Good catch 🥇 . Not yet, but I will have to add this functionality for exceptions https://github.com/getsentry/sentry/issues/50710